### PR TITLE
frontend/fix: #296 Added Ui api types and Flow for ZoneOtpRide

### PR DESCRIPTION
**Pull request by Jai-01**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
1. Added Ui in HomeScreen for OtpRide
2. Added otpRide/start api and its types
3. Stitched flow from entering otp to endRide

Video Link:- https://drive.google.com/file/d/1mnMtGkiI6KkyxLzxHJV7Key_qKnE0WcN/view?usp=sharing

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-driver/src/Flow.purs [hunk 1]
```diff
@@ -686,6 +686,29 @@ homeScreenFlow = do
               void $ lift $ lift $ toggleLoader false
               else pure $ toast (getString ERROR_OCCURED_PLEASE_TRY_AGAIN_LATER)
           homeScreenFlow
+    GO_TO_START_ZONE_RIDE {otp, lat, lon} -> do
+      _ <- pure $ printLog "HOME_SCREEN_FLOW GO_TO_START_ZONE_RIDE" "."
+      _ <- pure $ printLog "Lat in Floww: " lat
```
Comment (hkmangla) [inline Frontend/ui-driver/src/Flow.purs:691]: remove uneccessary print logs

File: Frontend/ui-driver/src/Flow.purs [hunk 2]
```diff
@@ -686,6 +686,29 @@ homeScreenFlow = do
...
+      _ <- pure $ printLog "Lon in Floww: " lon
+      void $ lift $ lift $ toggleLoader true
+      startZoneRideResp <- lift $ lift $ Remote.otpRide "" (Remote.makeOTPRideReq otp (fromMaybe 0.0 (fromString lat)) (fromMaybe 0.0 (fromString lon))) -- driver's lat long during starting ride
```
Comment (hkmangla) [inline Frontend/ui-driver/src/Flow.purs:694]: Please don't use fromMaybe at all. cc: @witcher-shailesh

File: Frontend/ui-driver/src/Services/Endpoint.purs [hunk 3]
```diff
@@ -98,3 +98,6 @@ linkReferralCode dummyString = (getBaseUrl "") <> "/driver/linkReferralCode"
 
 getPerformance :: String -> String
 getPerformance dummyString = (getBaseUrl "") <> "/driver/performance"
+
+otpRide :: String -> String
+otpRide dummyRideOtp = (getBaseUrl "") <> "/driver/otpRide/start"
```
Comment (hkmangla) [inline Frontend/ui-driver/src/Services/Endpoint.purs:103]: What is dummyRide Otp here?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -344,6 +347,9 @@ eval (RideActiveAction activeRide) state = updateAndExit state { data {activeRid
 
 eval RecenterButtonAction state = continue state
 
+eval ZoneOtpAction state = do
+  continue state { props = state.props { enterOtpModal = true, rideOtp = "", enterOtpFocusIndex = 0, otpIncorrect = false } }
```
Comment (hkmangla) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:351]: why empty string, use Nothing if needed.

File: Frontend/ui-driver/src/Services/Backend.purs [hunk 5]
```diff
@@ -658,3 +658,20 @@ getPerformanceBT payload = do
     where
         errorHandler (ErrorPayload errorPayload) =  do
             BackT $ pure GoBack
+
+---------------------------------------- getPerformance ---------------------------------------------
```
Comment (hkmangla) [inline Frontend/ui-driver/src/Services/Backend.purs:662]: why get performance?

File: Frontend/ui-driver/src/Services/Backend.purs [hunk 6]
```diff
@@ -658,3 +658,20 @@ getPerformanceBT payload = do
...
+
+otpRide dummyRideOtp payload = do
+        headers <- getHeaders ""
+        withAPIResult (EP.otpRide dummyRideOtp) unwrapResponse $ callAPI headers ((OTPRideRequest dummyRideOtp payload))
+    where
+        unwrapResponse (x) = x
+
+makeOTPRideReq :: String -> Number -> Number -> OTPRideReq
+makeOTPRideReq otp lat lon = OTPRideReq {
+    "specialZoneOtpCode": otp,
```
Comment (hkmangla) [inline Frontend/ui-driver/src/Services/Backend.purs:672]: why are they string?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Services/Endpoint.purs (hunk 3 above)
Comment (Jai-01) [reply Frontend/ui-driver/src/Services/Endpoint.purs:103]: for lazy evaluation we need to give some arguments

File: Frontend/ui-driver/src/Services/APITypes.purs [hunk 7]
```diff
@@ -1202,3 +1202,66 @@ instance showGetGetPerformanceRes :: Show GetPerformanceRes where show = generic
...
+instance encodeOTPRideReq :: Encode OTPRideReq where encode = defaultEncode
+
+instance makeOTPRideReq :: RestEndpoint OTPRideRequest DriverRideRes where
+    makeRequest reqBody@(OTPRideRequest dummyRideOtp (OTPRideReq rqBody)) headers = defaultMakeRequest POST (EP.otpRide dummyRideOtp) headers reqBody
+    decodeResponse = decodeJSON
+    encodeRequest req = standardEncode req
+
+data OTPRideRequest = OTPRideRequest String OTPRideReq
+
+newtype DriverRideRes = DriverRideRes
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Services/APITypes.purs:1227]: @Jai-01  this response body we are already having, Can you use that?

Comment (Rohit4997) [reply Frontend/ui-driver/src/Services/APITypes.purs:1227]: Also can you add translations ?

**Pull request merged**